lib/verify/response.rb

```diff
@@ -76,19 +77,20 @@ def address_missing?
 
     def full_name
       first_name = most_recent_verified_value(attributes.fetch("firstNames"))
-      middle_name = most_recent_verified_value(attributes.fetch("middleNames"))
+      middle_name = most_recent_verified_value(attributes.fetch("middleNames"), required: false)
       surname = most_recent_verified_value(attributes.fetch("surnames"))
 
       [first_name, middle_name, surname].compact.join(" ")
     end
 
-    def most_recent_verified_value(attributes)
+    def most_recent_verified_value(attributes, required: true)
       return if attributes.blank?
 
-      attributes.sort_by { |attribute| attribute["from"].present? ? Date.strptime(attribute["from"], "%Y-%m-%d") : 0 }
+      most_recent_verified_attribute = attributes.sort_by { |attribute| attribute["from"].present? ? Date.strptime(attribute["from"], "%Y-%m-%d") : 0 }
         .reverse
         .find { |attribute| attribute["verified"] }
-        .fetch("value")
+      raise MissingResponseAttribute, "No verified value found" if required && most_recent_verified_attribute.nil?
+      most_recent_verified_attribute&.fetch("value")
```

Comment:
the `&` shouldn't be necessary here, right?

Reply:
also `&.fetch(key)` is equivalent to `[key]`

Reply:
Ah yes I put the `&.` in to allow for `required: false` returning nil there. I'll change it to `[key]`

Reply:
Hmm, `[key]` won't work for optional values where `most_recent_verified_attribute` is nil

Reply:
we probably still want `fetch("value")` I think. If the most recent "verified" field doesn't have a "value" we probably want it to noisily fail so we can investigate

Reply:
I've put the turnery back in to allow for nil optional values as well